store.go

```diff
@@ -2479,8 +2493,28 @@ func (s *store) DeleteLayer(id string) error {
 func (s *store) DeleteImage(id string, commit bool) (layers []string, err error) {
 	layersToRemove := []string{}
 	if err := s.writeToAllStores(func(rlstore rwLayerStore) error {
-		if s.imageStore.Exists(id) {
-			image, err := s.imageStore.Get(id)
+		// Perform delete image on primary imageStore but if image
+		// is not found on primary store then try delete on any additional
+		// write-able image store.
+		imageStores := []rwImageStore{s.imageStore}
+		imageStores = append(imageStores, s.rwImageStores...)
+		imageFound := false
+		primaryImageStore := true
+		for _, imageStore := range imageStores {
+			if !imageStore.Exists(id) {
+				primaryImageStore = false
+				continue
+			}
```

Comment:
If we go through this for loop 3 times, and the 2nd iteration we hit the continue , but the third we find it, imageFound will be set to true.  Is that valid in that circumstance?  (Stretching my store memory greatly).

Reply:
Yes that sounds valid.